app/front-end/src/features/editor/components/filebarView/filebarGroupItem.tsx

```diff
@@ -26,68 +28,111 @@ import { alpha, Box, IconButton, Typography, useTheme } from '@mui/material';
 export const FilebarGroupItem: React.FC<FileModel> = (file) => {
   const Theme = useTheme();
   const Workspace = useWorkspaceContext();
-  const { blocked } = useStatusContext();
+  const { blocked, unsaved } = useStatusContext();
 
   const { id, label } = file;
 
+  const [confirmAction, setConfirmAction] = useState<'tab' | 'close' | null>(null);
+
+  const [isConfirmDialogOpen, setIsConfirmDialogOpen] = useState(false);
+
+  const handleTabConfirm = useCallback(() => {
+    Workspace.fileStateUpdate(file);
+    Workspace.filesHistoryStateUpdate(file);
+    setIsConfirmDialogOpen(false);
+    setConfirmAction(null);
+  }, [file, Workspace]);
+
+  const handleCloseConfirm = useCallback(() => {
+    Workspace.filesHistoryStateUpdate(undefined, file);
+    setIsConfirmDialogOpen(false);
+    setConfirmAction(null);
+  }, [file, Workspace]);
+
+  const handleTabClick = useCallback(() => {
+    if (!blocked) {
+      if (unsaved && Workspace.file.id !== id) {
+        setConfirmAction('tab');
+        setIsConfirmDialogOpen(true);
+      } else {
+        Workspace.fileStateUpdate(file);
+        Workspace.filesHistoryStateUpdate(file);
+      }
+    }
+  }, [blocked, unsaved, file, Workspace]);
+
+  const handleCloseIconClick = (event: React.MouseEvent) => {
+    event.stopPropagation();
+    if (!blocked) {
+      if (unsaved) {
```

Comment:
New issue. When closing the file which is not the current I have unsaved changes, I get the prompt of unsaved changes. It should let me close not relevant files. Should also check if it's the current file, otherwise don't prompt the unsaved changes

Reply:
Might be also correlated to switching between file in file history. Maybe the parent button of selecting different file is also invoked when pressing close button

Reply:
Great catch! Fixed.